### PR TITLE
Json serialization of the matrix metadata

### PR DESCRIPTION
**Pull request by jsaarimaa**
- Refactored the MetaProperty to have content specific subtypes.
- Updated the various utility methods to use and benefit from the new structure.
- The value type of each px entry can now be set via the configuration object.
- Implemented Json serialization for the MatrixMetadata objects.
- IReadOnlyMatrixMetadata can aso be used for serialization.
- Implemented deserialization for the MatrixMetadata objects.
- Implemented costom json converters for all of the classes that it was required in order to get polymorphic serialization and deserialization to work.
- Added unit tests for all of the converted utility methods related to the MetaProperty refactoring.
- Added unit tests for MatrixMetadata serialization / deserialization.
- Added unit tests for all of the custom converters.

### CODE REVIEW

File: Px.Utils.UnitTests/ModelTests/ExtensionTests/PropertyExtensionTests/CleanDringDelimeterTests.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+﻿using Px.Utils.Models.Metadata.ExtensionMethods;
+
+namespace Px.Utils.UnitTests.ModelTests.ExtensionTests.PropertyExtensionTests
+{
+    [TestClass]
+    public class CleanDringDelimeterTests
```
Comment (sakari-malkki) [inline Px.Utils.UnitTests/ModelTests/ExtensionTests/PropertyExtensionTests/CleanDringDelimeterTests.cs:6]: This is a typo I suppose?

File: Px.Utils/Models/Metadata/Dimensions/ContentDimensionValue.cs [hunk 2]
```diff
@@ -63,6 +65,30 @@ public ContentDimensionValue(
...
+        /// </summary>
+        /// <param name="code">Unique code among the values of this dimension</param>
+        /// <param name="name">Multilanguage name of the dimension value</param>
+        /// <param name="isVirtual">Indicates whether the value is in the original data or was created by computing or aggregating other values</param>
+        /// <param name="additionalProperties">Additional properties of the dimension value</param>
+        [JsonConstructor]
+        public ContentDimensionValue(
+            string code,
+            MultilanguageString name,
+            MultilanguageString unit,
```
Comment (sakari-malkki) [inline Px.Utils/Models/Metadata/Dimensions/ContentDimensionValue.cs:80]: unit, lastUpdated and precision are not documented in the summary params above

Comment (jsaarimaa) [reply Px.Utils/Models/Metadata/Dimensions/ContentDimensionValue.cs:83]: Documentation added.

File: Px.Utils.UnitTests/ModelTests/ExtensionTests/PropertyExtensionTests/CleanDringDelimeterTests.cs (hunk 1 above)
Comment (jsaarimaa) [reply Px.Utils.UnitTests/ModelTests/ExtensionTests/PropertyExtensionTests/CleanDringDelimeterTests.cs]: Renamed the file and the class.

**Review by sakari-malkki — APPROVED** [review]

**Pull request merged**